hazelcast/src/main/java/com/hazelcast/internal/management/request/GetMapEntryRequest.java

```diff
@@ -48,17 +48,16 @@ public int getType() {
 
     @Override
     public void writeResponse(ManagementCenterService mcs, JsonObject root) throws Exception {
-        IMap map = mcs.getHazelcastInstance().getMap(mapName);
+        root.add("result", writeResult(mcs.getHazelcastInstance().getMap(mapName)));
+    }
+    
+    private <K, V> JsonObject writeResult(IMap<K, V> map) throws Exception {
         JsonObject result = new JsonObject();
-        EntryView entry = null;
-        if (type.equals("string")) {
-            entry = map.getEntryView(key);
-        } else if (type.equals("long")) {
-            entry = map.getEntryView(Long.valueOf(key));
-        } else if (type.equals("integer")) {
-            entry = map.getEntryView(Integer.valueOf(key));
-        }
-        if (entry != null) {
+        if (!map.isEmpty()) {
+            @SuppressWarnings("unchecked")
+            Class<K> keyType = (Class<K>) map.keySet().iterator().next().getClass();
```

Comment:
`map.keySet()` will fetch the whole keyset. That can exhaust the network and memory if map is big enough.

Reply:
I was worried it might be an issue. But I did not see another way to find out what types of keys we use. Any better idea?

Reply:
`map.localKeySet()` could be an option. If there are no local keys (yet), that's unfortunate, but will not kill the cluster.

Reply:
If the type hint (string, long etc) is still there, this can be the fallback option on the UI (like _string_, _long_, ..,  _other_) and when other is selected you can resort to `map.localKeySet()` option. 

Like David said, there might be no local keys yet (or never if it's a lite member). Also, key class may not present on server too, but in that case there's nothing to do.

Reply:
As an aside: Is a "lite member" a member that does not store data? In that case I wonder why we would use this way of fetching data that is available though cluster API instead of letting MC become a lite member and just using the API there. We will have the same question of how to actually make this key/type problem work but that code would be in MC and we can drop the overhead of coding the communication explicitly for every case where we want cluster data.

@mdogan says there will always be cases where we cannot make it work if the user just supplies a `String`. The entire entry browser is a best effort functionality - I just thought we can make it a bit more versatile any easier to use.